CryptoPrices/Sources/MyCoin/Sources/MyCoin/MyCoinView/MyCoinViewModel.swift

```diff
@@ -33,20 +33,19 @@ import UseCaseProtocol
         }
     }
 
-    // TODO: Support different filters
-    public func fetchChartPricesData(id: String) async {
+    public func fetchChartPricesData(id: String, timeFrameItem: TimeFrameItem) async {
         do {
-            let dataPoints = try await getChartPricesUseCase.execute(coinID: id, filter: .oneDay)
+            let dataPoints = try await getChartPricesUseCase.execute(coinID: id, filter: timeFrameItem.timeFrame)
             chartData = dataPoints.map { ChartDataPointUIModel(dataPoint: $0) }
```

Comment:
Suggestion: Since 5 different filters will introduce a lot of chart data points, it feels inefficient that we just reload the chart every time from the server and discard the previous chart filter data. 

=> We should apply some caching mechanism here and only load the chart from the server the first time selecting a filter so that when the user switches the filters fast, we won't spam the requests to the server and improve the rendering time in a session. We will also have the capability to refresh later on so we can clean up the data at that time. What do you think? 🙏

FYI, here are some options to cache the chart data for us to consider. Let me know your thought 🙏:
  - Create a dictionary with the key type is `TimeFrameItem` and the value type is `[ChartDataPointUIModel]`.  This will allow storing the data for all filters directly in this `MyCoinViewModel` as an in-memory option.
  - Set up a local database like `Realm` or `CoreData` to store/retrieve the chart data accordingly. This way we can keep the cache data alive across sessions but required some forefront setups. This can be done in a separate PR if we want to follow this approach.

Reply:
Thank you for the suggestion 🙏 . It's a good point to apply some caching mechanisms to avoid spamming requests the server 👍, especially when the user quickly changes the timeframe for any purpose. But in crypto, the data changes so frequently (I mean in a second), caching the data is not quite the best practice. I also checked the timeframe bar on the [coingecko](https://www.coingecko.com/), they do send the request every time we change the timeframe, even though the chart shows not much difference :D. Therefore, I think we can keep this current implementation, even though our app only uses the Free API for not real-time data 🙏 . What do you think?

> - Create a dictionary with the key type is TimeFrameItem and the value type is [ChartDataPointUIModel]. This will allow storing the data for all filters directly in this MyCoinViewModel as an in-memory option.
> - Set up a local database like Realm or CoreData to store/retrieve the chart data accordingly. This way we can keep the cache data alive across sessions but required some forefront setups. This can be done in a separate PR if we want to follow this approach.

Again, thank you so much for the suggestion along with the solutions 👍 . Great suggestion.

Reply:
Ohhh, I don't think we are displaying a real-time chart and price here or anything, so that is why I suggest caching and I believe it would not be a bad call. But yeah, this is in the end a suggestion and we can have different views about this.

In fact, caching is indeed more of a complex approach as there can be extra things that need to be decided further down the road such as managing the expiration of the cache, auto-refreshing cache data, selecting cache type (in-memory, local database), etc. Thus, reloading the chart after every filter change is a much simpler implementation logic to deal with 👍

-> Let's keep the current implementation for now until we found it extra annoying when it reaches the API limits too often 🙏

Reply:
>In fact, caching is indeed more of a complex approach as there can be extra things that need to be decided further down the road such as managing the expiration of the cache, auto-refreshing cache data, selecting cache type (in-memory, local database), etc.

Nice 👍 .

Reply:
As @nkhanh44 mentioned, usually, for time-sensitive applications like stock market price, voting, etc, the data is updated so frequently so caching does not mean much IMHO. However, I think even for that kind of application, we won't update the data on the user's demand (like switching to another time filter or pulling to refresh) or time interval by making multiple network requests (polling). Instead (at least in my experience) the app maintains a single connection with the server through WebSocket 🙏 

It's still possible to implement effective caching but we would do that at the lower layer like the network. It's because the network layer can access the `HTTPResponse` which contains the information about data invalidation like `Cache-Control`, `Last-Modified`, ...

But as @minhnimble mentioned, we shouldn't bother with this complexity much at this point, so keeping the current implementation is fine for me too 🙏